### ISSUE TITLE
Property naming rule reports anonymous object's instance variables should use the screaming snake case

### ISSUE THREAD

**Issue by muniere** [pinterest/ktlint#2066]
## Expected Behavior

`ktlint` should allows the camelCase properties in anonymous objects, for example,

```kotlin
interface MyInterface {
    fun myFunction()
}

fun main() {
    val myObject = object : MyInterface {
        private val name = "Example"

        override fun myFunction() {
            print("Hello, $name")
        }
    }
}
```

## Observed Behavior

`ktlint` command fails with following output.

```
/tmp/MyInterface.kt:7:21: Property name should use the screaming snake case notation when the value can not be changed (standard:property-naming)

Summary error count (descending) by rule:
  standard:property-naming: 1
```

## Steps to Reproduce

Put the above code at `/tmp/MyInterface.kt`, and run the following command:

```
$ ktlint --experimental /tmp/MyInterface.kt
```

## Your Environment

- Version of ktlint used: 0.49.1
- Operating System and version: macOS Ventura 13.4

**Comment by paul-dingemans** [comment]
Both https://kotlinlang.org/docs/coding-conventions.html#property-names and https://developer.android.com/kotlin/style-guide#constant_names are describing the rules regarding constants:

Kotlin coding conventions:

> Names of constants (properties marked with const, or top-level or object val properties with no custom get function that hold deeply immutable data) should use uppercase underscore-separated ([screaming snake case](https://en.wikipedia.org/wiki/Snake_case)) names

Android style guide:

> Constants are val properties with no custom get function, whose contents are deeply immutable, and whose functions have no detectable side-effects. This includes immutable types and immutable collections of immutable types as well as scalars and string if marked as const. If any of an instance’s observable state can change, it is not a constant. Merely intending to never mutate the object is not enough.
> ...
> Constant values can only be defined inside of an object or as a top-level declaration. Values otherwise meeting the requirement of a constant but defined inside of a class must use a non-constant name.

So the way I read above, the `name` property is a constant.

**Comment by muniere** [comment]
Thanks for reply, Paul.

I reconsidered the `name` filed in anonymous object, and concluded that I should define it as a constant as you pointed out, with following reasons,

- the field doesn't need to be an instance field, but can be static field
- its value can be defined at compile time, not runtime

So, I close this issue and change my code with `companion object` !

Regards,
